effekt/shared/src/main/scala/effekt/Intelligence.scala

```diff
@@ -140,43 +140,48 @@ trait Intelligence {
     innerType = hole.innerType.map { t => pp"${t}" }
     expectedType = hole.expectedType.map { t => pp"${t}" }
   } yield {
-    val BindingInfo(importedTerms, importedTypes, terms, types) = allBindings(scope)
-    HoleInfo(hole.name.name, hole.decl.span, innerType, expectedType, importedTerms.toList.distinct, importedTypes.toList.distinct, terms.toList.distinct, types.toList.distinct)
+    val scopeInfo = allBindings(scope)
+    HoleInfo(
+      hole.name.name,
+      hole.decl.span,
+      innerType,
+      expectedType,
+      scopeInfo
+    )
   }
 
-  def allBindings(scope: Scope)(using C: Context): BindingInfo =
+  def allBindings(scope: Scope)(using C: Context): ScopeInfo =
     scope match {
       case Scope.Global(imports, bindings) =>
-        val (te1, ty1) = allBindings(imports)
-        val (te2, ty2) = allBindings(bindings)
-        BindingInfo(te1, ty1, te2, ty2)
+        val bindingsOut = allBindings(BindingOrigin.Imported, imports)
+          ++ allBindings(BindingOrigin.Defined, bindings)
+        ScopeInfo(None, ScopeKind.Global, bindingsOut.toList, None)
```

Comment:
I'm a bit reluctant about this change -- is the repeated `++` on iterables here OK, especially perf-wise? 👀